src/plugins/intel_gpu/tests/test_cases/dft_gpu_test.cpp

```diff
@@ -0,0 +1,1591 @@
+struct dft_gpu_test : public testing::TestWithParam<P> {
+public:
+    void test() {
+        auto p = testing::TestWithParam<P>::GetParam();
+        auto& engine = get_test_engine();
+
+        const auto input_format = format::get_default_format(p.input_shape.size());
+        const layout data_layout(type_to_data_type<T>::value, input_format, tensor(input_format, p.input_shape));
+        auto input = engine.allocate_memory(data_layout);
+        set_values(input, convert<T>(p.input_values));
+
+        topology topology;
+        topology.add(input_layout("input", input->get_layout()));
+        topology.add(dft("dft", "input", std::move(p.axes), p.output_shape, getKind<P>()));
+
+        network network(engine, topology);
+        network.set_input_data("input", input);
+        const auto outputs = network.execute();
+
+        EXPECT_EQ(outputs.size(), size_t(1));
+        EXPECT_EQ(outputs.begin()->first, "dft");
+
+        auto output = outputs.at("dft").get_memory();
+        cldnn::mem_lock<T> output_ptr(output, get_test_stream());
+
+        const auto expected_values = convert<T>(p.expected_values);
+        ASSERT_EQ(output_ptr.size(), expected_values.size());
+        for (size_t i = 0; i < output_ptr.size(); ++i) {
+            EXPECT_NEAR(expected_values[i], output_ptr[i], getThreshold());
+        }
+    }
+
+    static std::string PrintToStringParamName(const testing::TestParamInfo<P>& info) {
+        auto& p = info.param;
+        std::ostringstream result;
+        result << "InputShape=" << vec2str(p.input_shape) << "_";
+        result << "OutputShape=" << vec2str(p.output_shape) << "_";
+        result << "Precision=" << data_type_traits::name(type_to_data_type<T>::value) << "_";
+        result << "Axes=" << vec2str(p.axes) << "_";
+        result << "Inverse=" << (getKind<P>() == dft_kind::inverse);
+        if (!p.test_name.empty()) {
+            result << "_TestName=" << p.test_name;
+        }
+        return result.str();
+    }
+
+    float getThreshold() const;
+};
+
+template <>
+float dft_gpu_test<dft_test_params, float>::getThreshold() const {
+    return 1e-4f;
+}
+
+template <>
+float dft_gpu_test<dft_test_params, half_t>::getThreshold() const {
+    return 4e-2f;
+}
+
+template <>
+float dft_gpu_test<idft_test_params, float>::getThreshold() const {
+    return 4e-6f;
+}
+
+template <>
+float dft_gpu_test<idft_test_params, half_t>::getThreshold() const {
+    return 5e-4f;
+}
+
+using dft_gpu_test_f32 = dft_gpu_test<dft_test_params, float>;
+using dft_gpu_test_f16 = dft_gpu_test<dft_test_params, half_t>;
+using idft_gpu_test_f32 = dft_gpu_test<idft_test_params, float>;
+using idft_gpu_test_f16 = dft_gpu_test<idft_test_params, half_t>;
+
+TEST_P(dft_gpu_test_f32, dft_f32) {
+    test();
+}
+
+TEST_P(dft_gpu_test_f16, dft_f16) {
+    test();
+}
+
+TEST_P(idft_gpu_test_f32, idft_f32) {
+    test();
+}
+
+TEST_P(idft_gpu_test_f16, idft_f16) {
+    test();
+}
+
+const std::vector<float> input_data = {
```

Comment:
Looks like this blob vectors would be better to store in separate file and include here using `#include`, because it make this file bigger and harder to read the logic in `dft_gpu_test`
If you want to see in this scope `const std::vector<float> input_data` then it is better as for me to declare it like this `extern const std::vector<float> input_data`:
```cpp
extern const std::vector<float> input_data;
#include "dft_gpu_test_blobls.hpp"
```

Reply:
Yeah, nice idea. I took those tests data from ngraph reference tests, and also was thinking about extracting them somewhere. But agreed with @tgubanova-lohika that we won't do it for now as it will cause dependencies if we extract them together with ngraph and won't make sense to extract only in cldnn.